internal/cosmetic/addrule.go

```diff
@@ -0,0 +1,41 @@
+package cosmetic
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+)
+
+var (
+	// CosmeticRuleRegex matches cosmetic rules.
+	CosmeticRuleRegex = regexp.MustCompile(`^(?:([^#$]+?)##|##)(.+)$`)
```

Comment:
IMO this should be renamed to `RuleRegex` since both internal and external users of the variable could imply its purpose from the context. Repeating the package name within a variable is generally discouraged.

Reply:
Good point, and I agree. However, I have decided to keep the `cosmeticRuleRegex` variable in `filter.go` unchanged to maintain a consistent style.